tests/suites/test_suite_pk.function

```diff
@@ -2196,3 +2246,242 @@ exit:
+                                   mbedtls_test_rnd_std_rand, NULL), 0);
+    }
+
+    TEST_EQUAL(mbedtls_pk_verify(&pk_priv, md_for_test, in_buf, in_buf_len,
+                                 out_buf, out_buf_len), 0);
+    TEST_EQUAL(mbedtls_pk_verify(&pk_pub, md_for_test, in_buf, in_buf_len,
+                                 out_buf, out_buf_len), 0);
+
+    if (PSA_ALG_IS_RSA_OAEP(key_alg) || PSA_ALG_IS_RSA_PSS(key_alg)) {
+        TEST_EQUAL(mbedtls_pk_verify_ext(MBEDTLS_PK_RSASSA_PSS, &pss_opt,
+                                         &pk_priv, md_for_test, in_buf, in_buf_len,
+                                         out_buf, out_buf_len), 0);
+        TEST_EQUAL(mbedtls_pk_verify_ext(MBEDTLS_PK_RSASSA_PSS, &pss_opt,
+                                         &pk_pub, md_for_test, in_buf, in_buf_len,
+                                         out_buf, out_buf_len), 0);
+    }
+
+#if defined(MBEDTLS_PSA_UTIL_HAVE_ECDSA)
+    if (PSA_ALG_IS_SIGN(key_alg) && is_psa_hash_alg_specified) {
+        /* ECDSA signature requires PK->PSA format conversion. */
+        if (PSA_ALG_IS_ECDSA(key_alg)) {
+            TEST_EQUAL(mbedtls_ecdsa_der_to_raw(mbedtls_pk_get_bitlen(&pk_pub),
+                                                out_buf, out_buf_len, out_buf,
+                                                sizeof(out_buf), &out_buf_len), 0);
+        }
+        PSA_ASSERT(psa_verify_hash(pub_key_id, key_alg, in_buf, in_buf_len,
+                                   out_buf, out_buf_len));
+    }
+
+    /* Test #3: check sign/verify interoperability also in the opposite direction:
+     * sign with PSA and verify with PK. Key's policy must include a valid hash
+     * algorithm (not any).
+     */
+    if (PSA_ALG_IS_SIGN(key_alg) && is_psa_hash_alg_specified) {
+        PSA_ASSERT(psa_sign_hash(priv_key_id, key_alg, in_buf, in_buf_len,
+                                 out_buf, sizeof(out_buf), &out_buf_len));
+        /*  ECDSA signature requires PSA->PK format conversion */
+        if (PSA_ALG_IS_ECDSA(key_alg)) {
+            TEST_EQUAL(mbedtls_ecdsa_raw_to_der(mbedtls_pk_get_bitlen(&pk_pub),
+                                                out_buf, out_buf_len, out_buf,
+                                                sizeof(out_buf), &out_buf_len), 0);
+        }
+        TEST_EQUAL(mbedtls_pk_verify(&pk_pub, md_for_test, in_buf, in_buf_len,
+                                     out_buf, out_buf_len), 0);
+    }
+#endif /* MBEDTLS_PSA_UTIL_HAVE_ECDSA */
+
+    /* Test #4: in case of RSA key pair try also encryption/decryption.
+     * Hash algorithm does not matter when padding mode is PKCS1 v1.5, whereas
+     * it must be not null or any in case of v2.1 (OAEP). */
+    int test_encryption = (PSA_ALG_IS_RSA_PKCS1V15_SIGN(key_alg) ||
+                           (key_alg == PSA_ALG_RSA_PKCS1V15_CRYPT) ||
+                           is_psa_hash_alg_specified);
+    if ((key_type == PSA_KEY_TYPE_RSA_KEY_PAIR) && test_encryption) {
```

Comment:
If I remember correctly this extra check was needed because there were some test failing using the ANY_HASH alg. However as you pointed out in a [later comment](https://github.com/Mbed-TLS/mbedtls/pull/8774#discussion_r1499891284) that's probably due to some wrong configuration. I guess that removing that case will help also optimizing this check here.